src/foam/nanos/crunch/predicate/CapabilityGranted.js

```diff
@@ -32,17 +42,19 @@ foam.CLASS({
         if ( ! ( obj instanceof X ) ) return false;
         var x = (X) obj;
 
-        var newUCJ = (UserCapabilityJunction) x.get("NEW");
-        if ( newUCJ != null ) {
-          x = x.put("subject", newUCJ.getSubject(x));
+        if ( getSubjectFromUCJ() ) {
+          var newUCJ = (UserCapabilityJunction) x.get("NEW");
+          if ( newUCJ != null ) {
+            x = x.put("subject", newUCJ.getSubject(x));
```

Comment:
your getting a UCJ, and getting the subject from it??

Reply:
If a UCJ rule or predicate is executed within a system context it will need to know what subject the UCJ applies to. The logic works like this:

```
associatedEntity=0 -> subject is the source id alone
associatedEntity=1 -> subject is the source id alone
associatedEntity=2 -> subject is the realUser + effectiveUser
```

I noticed this logic was already present in a few places so I added `getSubject` to UserCapabilityJunction. It's basically asking "what's the most specific subject we can get with respect to this UCJ?"

The way the system works right now seems to be described pretty well by:
- _checking_ an associatedEntity=2 UCJ, you can get information the user and business's UCJs
- _affecting_ other UCJ's should happen the other way - a change to a business UCJ can affect any associatedEntity=2 UCJ, but it should never happen the other way around.